### PR TITLE
flake8 upgrade to 3.8.3

### PR DESCRIPTION
**Pull request by pratik-vii**
### What was wrong?

Related to Issue #1720

### How was it fixed?
* upgraded flake8 to 3.8.3
* W503 ignored where required.

#### Cute Animal Picture

Koala!!

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -51,10 +51,10 @@
 }
 
 extras_require['dev'] = (
-    extras_require['tester'] +
-    extras_require['linter'] +
-    extras_require['docs'] +
-    extras_require['dev']
+    extras_require['tester']
+    + extras_require['linter']  # noqa: W503
```
Comment (pipermerriam) [inline setup.py:55]: I would be ok with adding `W503` to the ignored warnings in our `tox.ini` file.

Comment (pratik-vii) [reply setup.py:55]: yeah. right! will do

**Review by pipermerriam — APPROVED** [review]
Thanks for doing this cleanup for us!  Would you be up for adding W503 to the ignored linting codes so we can drop a few of the `# noqa: W503` comments?

**Comment by pipermerriam** [comment]
(also, you should work on getting CI passing)

**Comment by pratik-vii** [comment]
> (also, you should work on getting CI passing)

some of the CI will fail because of [Known issue](https://github.com/ipfs-shipyard/py-ipfs-http-client/issues/239) with Infura IPFS endpoint

https://github.com/ethereum/web3.py/pull/1757#issuecomment-702628285

**Comment by kclowes** [comment]
@pratik-vii thanks for taking this on! I just pulled in the most recent master that should have the tests fixed, so as soon as those pass, I'll pull this in. :tada:

**Pull request merged**